Jenkinsfile

```diff
@@ -1,51 +1,58 @@
-                            archiveArtifacts allowEmptyArchive: true, artifacts: '**/target/site/*.zip'
-                        }
-                    }
-                }
-                stage('Test (Slow Tests)') {
-                    environment {
-                        TEST_MODE = 'slow'
-                    }
-                    steps {
-                        sh 'export DATADIR=$(pwd)/toscana-data2 && rm -r -f $(pwd)/toscana-data2 && mkdir $(pwd)/toscana-data2 && mvn test -pl core'
-                    }
-                    post {
-                        always {
-                            junit(testResults: 'core/target/surefire-reports/*.xml', allowEmptyResults: true)
-                            sh 'cd core/target/site && zip -r coverage-core-slow.zip jacoco && cd ../../../; exit 0'
-                            archiveArtifacts allowEmptyArchive: true, artifacts: '**/target/site/*.zip'
-                        }
-                    }
+        stage('Test') { 
+            parallel { 
+                stage('Test (Fast Tests)') { 
+                    environment { 
+                        TEST_MODE = 'fast' 
+                    } 
+                    steps { 
+                        sh 'export DATADIR=$(pwd)/toscana-data && rm -rf $(pwd)/toscana-data && mkdir $(pwd)/toscana-data && mvn test -B' 
+                    } 
+                    post { 
+                        always { 
+                            junit(testResults: '**/target/surefire-reports/*.xml', allowEmptyResults: true) 
+                            sh '[ -d core/target/site ] && cd core/target/site && zip -r coverage-core-fast.zip jacoco && cd -; exit 0' 
+                            sh '[ -d core/target/site ] && cd cli/target/site && zip -r coverage-cli-fast.zip jacoco && cd -; exit 0' 
+                            archiveArtifacts allowEmptyArchive: true, artifacts: '**/target/site/*.zip' 
+                        } 
+                    } 
+                } 
+                stage('Test (Slow Tests)') { 
+                    environment { 
+                        TEST_MODE = 'slow' 
+                    } 
+                    steps { 
+                        sh 'export DATADIR=$(pwd)/toscana-data2 && rm -rf $(pwd)/toscana-data2 && mkdir $(pwd)/toscana-data2 && mvn test -B' 
+                    } 
+                    post { 
+                        always { 
+                            junit(testResults: 'core/target/surefire-reports/*.xml', allowEmptyResults: true) 
+                            sh '[ -d core/target/site ] && cd core/target/site && zip -r coverage-core-slow.zip jacoco && cd -; exit 0' 
+                            archiveArtifacts allowEmptyArchive: true, artifacts: '**/target/site/*.zip' 
+                        } 
+                    } 
                 }
             }
-        }
-        stage('Cleanup') {
+        } 
+        stage('Deploy') {
             steps {
```

Comment:
Maybe add a When conditon here to only execute this if the Worker only has a specific label.

This will prevent the failiure of the build on systems that do not want do deploy automatically

Reply:
fixed